bundlegen/cli/main.py

```diff
@@ -104,15 +105,24 @@ def generate(image, outputdir, platform, searchpath, creds, ipk, yes):
 
     # Get the app metadata as a dictionary
     app_metadata_dict = {}
-    appmetadata = os.path.join(outputdir, "rootfs", "appmetadata.json")
-    if os.path.exists(appmetadata):
-        with open(appmetadata) as metadata:
+    app_metadata_file = os.path.join(outputdir, "rootfs", "appmetadata.json")
+    if os.path.exists(app_metadata_file):
+        with open(app_metadata_file) as metadata:
             app_metadata_dict = json.load(metadata)
         # remove app metadata from image rootfs
-        os.remove(appmetadata)
+        os.remove(app_metadata_file)
     else:
```

Comment:
This will mean that if app_metadata_file exists in rootfs then even if user wants to use other appmetadata (for whatever reasons) then they will be ignored. If we want to give user possibility to overwrite appmetadata then maybe change condition orders (i.e. check if appmetadata exists and if not then check for app_metadata_file).
But maybe this is intentional, if so please document that this parameter will be skipped when app_metadata_file is inside of rootfs. And maybe even print some warning that this situation occurred (user wanted to change metadata but it was skipped).